Networking v2 Trunk support - List

celebdor opened the pull request:
This patch adds the Networking extension trunk support for List
operations.

Signed-off-by: Antoni Segura Puimedon <celebdor@gmail.com>

For #1257 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron/blob/master/neutron/services/trunk/plugin.py#L189-L196
https://github.com/openstack/neutron/blob/master/neutron/services/trunk/models.py#L26-L49

celebdor commented:
recheck

jtopjian commented:
@celebdor There appears to be an issue with OpenLab at the moment. Since the OpenLab devstack configuration does not support trunks, the acceptance tests for trunks are not being run, so you can ignore the failures.

https://github.com/gophercloud/gophercloud/pull/1262

jtopjian commented:
Just be aware that, at this time, you can't be dependent on OpenLab for validating your changes.

celebdor commented:
Acceptance test being run on my devstack environment:
`[stack@kuryr-a32c-02 gophercloud]$ git checkout trunk_list
Previous HEAD position was 7566894... Networking v2 Trunk support - Create
Branch trunk_list set up to track remote branch trunk_list from celebdor.
Switched to a new branch 'trunk_list'
[stack@kuryr-a32c-02 gophercloud]$ git show
commit bdc94b75b2c859f30e5f9ee82f89fb9e945dc456
Author: Antoni Segura Puimedon <celebdor@gmail.com>
Date:   Thu Oct 4 15:31:03 2018 +0200

    Networking v2 Trunk support - List
    
    This patch adds the Networking extension trunk support for List
    operations.
    
    Signed-off-by: Antoni Segura Puimedon <celebdor@gmail.com>

diff --git a/acceptance/openstack/networking/v2/extensions/trunks/trunks_test.go b/acceptance/openstack/n
index f898ef2..3b76fa0 100644
--- a/acceptance/openstack/networking/v2/extensions/trunks/trunks_test.go
+++ b/acceptance/openstack/networking/v2/extensions/trunks/trunks_test.go
@@ -8,6 +8,7 @@ import (
        "github.com/gophercloud/gophercloud/acceptance/clients"
        "github.com/gophercloud/gophercloud/acceptance/openstack/networking/v2"
        "github.com/gophercloud/gophercloud/acceptance/tools"
+       "github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/trunks"
 )
 
 func TestTrunkCRUD(t *testing.T) {
@@ -57,3 +58,24 @@ func TestTrunkCRUD(t *testing.T) {
 
        tools.PrintResource(t, trunk)
 }
+
+func TestTrunkList(t *testing.T) {
+       client, err := clients.NewNetworkV2Client()
+       if err != nil {
+               t.Fatalf("Unable to create a network client: %v", err)
+       }
+
+       allPages, err := trunks.List(client, nil).AllPages()
+       if err != nil {
+               t.Fatalf("Unable to list trunks: %v", err)
+       }
+
+       allTrunks, err := trunks.ExtractTrunks(allPages)
+       if err != nil {
+               t.Fatalf("Unable to extract trunks: %v", err)
+       }
+
+       for _, trunk := range allTrunks {
+               tools.PrintResource(t, trunk)
+       }
+}
diff --git a/openstack/networking/v2/extensions/trunks/doc.go b/openstack/networking/v2/extensions/trunks
index 6d563de..79dd0a5 100644
--- a/openstack/networking/v2/extensions/trunks/doc.go
[stack@kuryr-a32c-02 gophercloud]$ go test -v -tags "fixtures acceptance" ./acceptance/openstack/networking/v2/extensions/trunks/
=== RUN   TestTrunkCRUD
--- PASS: TestTrunkCRUD (24.86s)
	networking.go:32: Attempting to create network: TESTACC-SMtsF112
	networking.go:39: Successfully created network.
	networking.go:248: Attempting to create subnet: TESTACC-QzBfl2dT
	networking.go:255: Successfully created subnet.
	networking.go:80: Attempting to create port: TESTACC-QwPlKEfu
	networking.go:103: Successfully created port: TESTACC-QwPlKEfu
	networking.go:80: Attempting to create port: TESTACC-xRDkyy8e
	networking.go:103: Successfully created port: TESTACC-xRDkyy8e
	networking.go:80: Attempting to create port: TESTACC-4Pnc3dqJ
	networking.go:103: Successfully created port: TESTACC-4Pnc3dqJ
	trunks.go:30: Attempting to create trunk: TESTACC-r9oNrZkV
	trunks.go:33: Successfully created trunk
	tools.go:72: {
		  "status": "ACTIVE",
		  "sub_ports": [
		    {
		      "segmentation_id": 1,
		      "segmentation_type": "vlan",
		      "port_id": "879e6427-bcf2-4a03-a316-3286aeaa3453"
		    },
		    {
		      "segmentation_id": 2,
		      "segmentation_type": "vlan",
		      "port_id": "031fdc7d-d382-40f9-82db-679d2a43b821"
		    }
		  ],
		  "name": "TESTACC-r9oNrZkV",
		  "admin_state_up": true,
		  "project_id": "e153f3f9082240a5974f667cfe1036e3",
		  "tenant_id": "e153f3f9082240a5974f667cfe1036e3",
		  "created_at": "2018-10-09T07:25:55Z",
		  "updated_at": "2018-10-09T07:25:58Z",
		  "revision_number": 1,
		  "port_id": "7c9e53b3-abf1-49ec-bb2d-cb3faef565ec",
		  "id": "83b07ffc-0018-4aa8-a8bd-ffae1ae1c185",
		  "description": "Trunk created by gophercloud"
		}
	trunks.go:39: Attempting to delete trunk: 83b07ffc-0018-4aa8-a8bd-ffae1ae1c185
	trunks.go:45: Deleted trunk: 83b07ffc-0018-4aa8-a8bd-ffae1ae1c185
	networking.go:413: Attempting to delete port: 031fdc7d-d382-40f9-82db-679d2a43b821
	networking.go:420: Deleted port: 031fdc7d-d382-40f9-82db-679d2a43b821
	networking.go:413: Attempting to delete port: 879e6427-bcf2-4a03-a316-3286aeaa3453
	networking.go:420: Deleted port: 879e6427-bcf2-4a03-a316-3286aeaa3453
	networking.go:413: Attempting to delete port: 7c9e53b3-abf1-49ec-bb2d-cb3faef565ec
	networking.go:420: Deleted port: 7c9e53b3-abf1-49ec-bb2d-cb3faef565ec
	networking.go:427: Attempting to delete subnet: f9964368-bb6b-4e15-90a8-aadd91d679ee
	networking.go:434: Deleted subnet: f9964368-bb6b-4e15-90a8-aadd91d679ee
	networking.go:399: Attempting to delete network: 8e6cdebc-2234-4138-afa5-f9f36a42eda7
	networking.go:406: Deleted network: 8e6cdebc-2234-4138-afa5-f9f36a42eda7
=== RUN   TestTrunkList
--- PASS: TestTrunkList (0.24s)
	tools.go:72: {
		  "status": "ACTIVE",
		  "sub_ports": [
		    {
		      "segmentation_id": 11,
		      "segmentation_type": "vlan",
		      "port_id": "424da4b7-7868-4db2-bb71-05155601c6e4"
		    }
		  ],
		  "name": "mytrunk",
		  "admin_state_up": true,
		  "project_id": "e153f3f9082240a5974f667cfe1036e3",
		  "tenant_id": "e153f3f9082240a5974f667cfe1036e3",
		  "created_at": "2018-10-01T15:29:39Z",
		  "updated_at": "2018-10-01T15:43:04Z",
		  "revision_number": 3,
		  "port_id": "16c425d3-d7fc-40b8-b94c-cc95da45b270",
		  "id": "3e72aa1b-d0da-48f2-831a-fd1c5f3f99c2",
		  "description": ""
		}
PASS
ok  	github.com/gophercloud/gophercloud/acceptance/openstack/networking/v2/extensions/trunks	25.112s
`

celebdor commented:
I think it should be ready to merge

jtopjian commented:
> I meant if help was needed fixing the devstack deployment, since I'm the person in charge of devstack in openstack/kuryr-kubernetes, I could maybe lend a hand.

oh! Thanks for the offer, but I don't think so. I think OpenLab's devstack config just needs to enable the "trunks" extension. 

@liu-sheng @kiwik @yongfengdu would you be able to help with this? I believe I can add additional services to be tested, but I can't configure those services.
